lib/data/kinetic/backend.js

```diff
@@ -0,0 +1,157 @@
+import crypto from 'crypto';
+import kinetic from 'kineticlib';
+import stream from 'stream';
+import net from 'net';
+
+import { errors } from 'arsenal';
+
+import config from '../../Config';
+
+let sequence = 1;
+
+const kDrives = {
+    port: config.kinetic ? config.kinetic.port : 8123,
+    host: config.kinetic ? config.kinetic.host : 'localhost',
+};
+
+function propError(pdu) {
+    const statusCode = pdu.getStatusCode();
+    if (statusCode !== kinetic.errors.SUCCESS) {
+        if (statusCode === kinetic.errors.NOT_FOUND) {
+            return errors.ObjNotFound;
+        }
+        return pdu.getErrorMessage();
+    }
+    return undefined;
+}
+
+function newSocket(callback) {
+    const socket = new net.Socket({ allowHalfOpen: false }).pause();
+    socket.setKeepAlive(true);
+    socket.unref();
+    return socket.connect(kDrives.port, kDrives.host, () => {
+        kinetic.streamToPDU(socket, (err, pdu) => {
+            if (err) {
+                return callback(err);
+            }
+            const err1 = propError(pdu);
+            return callback(err1, socket);
+        });
+    });
+}
+
+function putKinetic(socket, value, options, callback) {
+    const key = crypto.randomBytes(20);
+    const obj = Buffer.concat(value);
+    const tag = crypto.createHmac("sha1", "asdfasdf").update(obj).digest();
+    const pdu = new kinetic.PutPDU(sequence, key, obj.length, tag, options);
+    ++sequence;
+
+    socket.write(Buffer.concat([pdu.read(), obj]), err => {
+        if (err) {
+            return callback(err);
+        }
+        kinetic.streamToPDU(socket, (err, pdu) => {
+            const err1 = propError(pdu);
+            socket.destroy();
+            return callback(err || err1, key);
+        });
+        return undefined;
+    });
+}
+
+function getKinetic(socket, pdu, callback) {
+    let count = 0;
+    const chunkSize = pdu.getChunkSize();
+    const value = [];
+    const err = propError(pdu);
+    socket.resume();
```

Comment:
You do not need to resume the socket here, since you put a ```socket.on('data')``` the socket will pass in the flowing mode automatically.
https://nodejs.org/dist/latest-v4.x/docs/api/stream.html#stream_class_stream_readable